streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamJoinWindow.java

```diff
@@ -27,15 +26,8 @@
 
     private final String windowName;
 
-    /**
-     * @throws TopologyException if retention period of the join window is less than expected
-     */
-    KStreamJoinWindow(String windowName, long windowSizeMs, long retentionPeriodMs) {
+    KStreamJoinWindow(final String windowName) {
         this.windowName = windowName;
-
-        if (windowSizeMs > retentionPeriodMs)
```

Comment:
Why do you want to remove this check?

Reply:
We don't need the check because I've removed `windowSizeMs` and `retentionPeriodMs`. They were unused.

Reply:
That said, we should still enforce this precondition at some point. I propose adding this check:
```

        if (windowSize > retentionPeriod) {
            throw new TopologyException("The retention period of the window store "
                + name + " must be no smaller than its window size. Got size=["
                + windowSize + "], retention=[" + retentionPeriod + "]");
        }
```
alongside the other precondition checks in 
in `org.apache.kafka.streams.state.Stores#persistentWindowStore(java.lang.String, long, long, boolean, long)`

Reply:
> They were unused.

Well, they are used in the check -- thus, my question was about where do we check this when we remove it :) Adding it to `Stores#persistentWindowStore` sounds good to me.

Reply:
Ah, I'm glad we're on the same page. Thanks for pointing it out.